### PR TITLE
Feature/jellyfish buffer

### PR DESCRIPTION
**Pull request by SpiderDan98**

### CODE REVIEW

File: bundles/jellyfish-buffer/tests/FondOfOryx/Zed/JellyfishBuffer/Business/Buffer/JellyfishBufferOrderTest.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * @return void
+     */
+    public function testBufferOrder(): void
+    {
+        $this->jellyfishBufferOrder->bufferOrder(
+            $this->jellyfishOrderMock,
+            $this->options
+        );
+    }
```
Comment (sourcecube) [inline bundles/jellyfish-buffer/tests/FondOfOryx/Zed/JellyfishBuffer/Business/Buffer/JellyfishBufferOrderTest.php:60]: What is the result of this test ?

File: bundles/jellyfish-sales-order-b2c/composer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "fond-of-spryker/jellyfish-sales-order-b2c",
+  "name": "fond-of-oryx/jellyfish-sales-order-b2c",
```
Comment (sourcecube) [inline bundles/jellyfish-sales-order-b2c/composer.json:2]: Is this necessary to name this b2c ? What about a plugin which describes the purpose of this bundle.

File: bundles/jellyfish-sales-order/LICENSE [hunk 3]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2019 FOND OF GmbH
```
Comment (sourcecube) [inline bundles/jellyfish-sales-order/LICENSE:3]: License update.

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/JellyfishSalesOrderBusinessFactory.php [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderExpenseMapper;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderExpenseMapperInterface;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderItemMapper;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderItemMapperInterface;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderMapper;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderMapperInterface;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderPaymentMapper;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderPaymentMapperInterface;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderTotalsMapper;
+use FondOfOryx\Zed\JellyfishSalesOrder\Business\Model\Mapper\JellyfishOrderTotalsMapperInterface;
```
Comment (sourcecube) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/JellyfishSalesOrderBusinessFactory.php:22]: Have you thought about grouping those use declarations ?

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/JellyfishSalesOrderBusinessFactory.php [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            $this->getOrderBeforeExportPlugins()
+        );
+    }
+
+    /**
+     * @return \FondOfOryx\Zed\JellyfishSalesOrder\Dependency\Service\JellyfishSalesOrderToUtilEncodingServiceInterface
+     */
+    protected function getUtilEncodingService(): JellyfishSalesOrderToUtilEncodingServiceInterface
+    {
+        return $this->getProvidedDependency(JellyfishSalesOrderDependencyProvider::SERVICE_UTIL_ENCODING);
```
Comment (sourcecube) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/JellyfishSalesOrderBusinessFactory.php:143]: This is not listed as dependency

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderExpenseMapper.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    /**
+     * @param \Orm\Zed\Sales\Persistence\SpySalesExpense $salesExpense
+     *
+     * @return \Generated\Shared\Transfer\JellyfishOrderExpenseTransfer
+     */
+    public function fromSalesExpense(SpySalesExpense $salesExpense): JellyfishOrderExpenseTransfer
+    {
+        $jellyfishOrderExpense = new JellyfishOrderExpenseTransfer();
+
+        $jellyfishOrderExpense->setType($salesExpense->getType())
```
Comment (julianzimmermann) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderExpenseMapper.php:19]: return

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderItemMapper.php [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            ->setUnitTaxAmount((int)round($salesOrderItem->getTaxAmount() / $quantity))
+            ->setUnitDiscountAmountAggregation((int)round($salesOrderItem->getDiscountAmountAggregation() / $quantity))
+            ->setUnitDiscountAmountFullAggregation((int)round($salesOrderItem->getDiscountAmountFullAggregation() / $quantity))
+            ->setSumTaxAmount($salesOrderItem->getTaxAmount())
+            ->setSumPrice($salesOrderItem->getPrice())
+            ->setSumPriceToPayAggregation($salesOrderItem->getPriceToPayAggregation())
+            ->setSumDiscountAmountAggregation($salesOrderItem->getDiscountAmountAggregation())
+            ->setSumDiscountAmountFullAggregation($salesOrderItem->getDiscountAmountFullAggregation());
+
+        $jellyfishOrderItemTransfer = $this->expandOrderItemTransfer($jellyfishOrderItemTransfer, $salesOrderItem);
```
Comment (julianzimmermann) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderItemMapper.php:50]: return

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderMapper.php [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+            ->setSystemCode($this->systemCode)
+            ->setPayments($this->mapSalesOrderToPayments($salesOrder))
+            ->setBillingAddress($this->mapSalesOrderToBillingAddress($salesOrder))
+            ->setShippingAddress($this->mapSalesOrderToShippingAddress($salesOrder))
+            ->setExpenses($this->mapSalesOrderToExpenses($salesOrder))
+            ->setDiscounts($this->mapSalesOrderToDiscounts($salesOrder))
+            ->setTotals($this->mapSalesOrderToTotals($salesOrder))
+            ->setCreatedAt($salesOrder->getCreatedAt()->format('Y-m-d H:i:s'));
+
+        $jellyfishOrderTransfer = $this->expandOrderTransfer($jellyfishOrderTransfer, $salesOrder);
```
Comment (julianzimmermann) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderMapper.php:101]: return

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderPaymentMapper.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * @param \Orm\Zed\Payment\Persistence\SpySalesPayment $salesPayment
+     *
+     * @return \Generated\Shared\Transfer\JellyfishOrderPaymentTransfer
+     */
+    public function fromSalesPayment(SpySalesPayment $salesPayment): JellyfishOrderPaymentTransfer
+    {
+        $jellyfishOrderPayment = new JellyfishOrderPaymentTransfer();
+
+        $jellyfishOrderPayment->setAmount($salesPayment->getAmount())
```
Comment (julianzimmermann) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderPaymentMapper.php:19]: return

File: bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderTotalsMapper.php [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+            return $jellyfishOrderTotals;
+        }
+
+        $lastOrderTotals = $salesOrder->getLastOrderTotals();
+
+        if (!$lastOrderTotals) {
+            return $jellyfishOrderTotals;
+        }
+
+        $jellyfishOrderTotals->setExpenseTotal($lastOrderTotals->getOrderExpenseTotal())
```
Comment (julianzimmermann) [inline bundles/jellyfish-sales-order/src/FondOfOryx/Zed/JellyfishSalesOrder/Business/Model/Mapper/JellyfishOrderTotalsMapper.php:29]: return

**Review by julianzimmermann — APPROVED** [review]

**Pull request merged**